other/upgrade.php

```diff
@@ -3590,7 +3591,466 @@ function smf_strtolower($string)
 		return mb_strtolower($string, 'UTF-8');
 	require_once($sourcedir . '/Subs-Charset.php');
 	return utf8_strtolower($string);
-};
+}
+
+/**
+ * Handles converting your database to UTF-8
+ */
+function convertUtf8()
+{
+	global $db_character_set, $sourcedir, $smcFunc, $modSettings, $language, $db_prefix, $db_type;
+
+	// First make sure they aren't already on UTF-8 before we go anywhere...
+	if ($db_type == 'postgresql' || ($db_character_set === 'utf8' && !empty($modSettings['global_character_set']) && $modSettings['global_character_set'] === 'UTF-8'))
+	{
+		echo 'Database already UTF-8. Skipping.';
```

Comment:
PostgreSQL doesn't have proper charset/collation support like MySQL does. You can only set the charset and collation at the database level, and only when the database is created. It does however have good support for automatic conversion between the client and the database charset. In most cases it will be UTF-8 already anyway (that's the default setting and cPanel doesn't give you an option to choose the charset or collation when creating the db).

Has anyone tested this to see how well it works? I don't have a good board to test this with at the moment, though I could theoretically copy my existing database and run tests on that.